File: app/src/main/java/com/adeebtechlab/apps/Share.java
package com.adeebtechlab.apps;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class Share extends AppCompatActivity {


    Button cur, fol, hom, dev, sha, b1, b11, b2, b22, b3, b33, b4, b44;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_share);


        hom = findViewById(R.id.hom);
        fol = findViewById(R.id.fol);
        dev = findViewById(R.id.dev);
        cur = findViewById(R.id.cur);


        b1 = findViewById(R.id.button1);
        b11 = findViewById(R.id.button11);
        b2 = findViewById(R.id.button2);
        b22 = findViewById(R.id.button22);
        b3 = findViewById(R.id.button3);
        b33 = findViewById(R.id.button33);
        b4 = findViewById(R.id.button4);
        b44 = findViewById(R.id.button44);



        //------------Button----------------------------------------

        hom.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(Share.this, MainActivity.class);
                startActivity(intent);

            }
        });


        //------------Button----------------------------------------

        fol.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(Share.this, Folder.class);
                startActivity(intent);

            }
        });

        //------------Button----------------------------------------

        dev.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(Share.this, Developer.class);
                startActivity(intent);

            }
        });


        //------------Button----------------------------------------

        cur.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(Share.this, Curses.class);
                startActivity(intent);

            }
        });





        //---------------------------Downloader-----Button---------------------------------
        b1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String url = "https://play.google.com/store/apps/details?id=com.adeebtechlab.apps";
                Intent i = new Intent(Intent.ACTION_VIEW);
                i.setData(Uri.parse(url));
                startActivity(i);
            }
        });




//---------------------------Share-----Button---------------------------------
        b11.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(Intent.ACTION_SEND);
                intent.setType("text/plain");
                String Body = "Download This App";
                String Sub = "Download Free Adeeb Technology Lab App:          https://play.google.com/store/apps/details?id=com.adeebtechlab.apps";
                intent.putExtra(Intent.EXTRA_TEXT, Body);
                intent.putExtra(Intent.EXTRA_TEXT, Sub);
                startActivity(Intent.createChooser(intent, "Share Using"));

            }
        });





        //---------------------------Downloader-----Button---------------------------------
        b2.setOnClickListener(new View.OnClickListener() {
                                  public void onClick(View view) {
                                      Toast.makeText(getApplicationContext(), "software coming soon", Toast.LENGTH_LONG).show();
                                  }
                              });




//---------------------------Share-----Button---------------------------------

            b22.setOnClickListener(new View.OnClickListener() {
                public void onClick(View view) {
                    Toast.makeText(getApplicationContext(), "software coming soon", Toast.LENGTH_LONG).show();
                }
            });




        //---------------------------Downloader-----Button---------------------------------
        b3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String url = "https://chromewebstore.google.com/detail/adeeb-technology-lab/hmgnckmfplfccnkinmmlmicpldfpnhme";
                Intent i = new Intent(Intent.ACTION_VIEW);
                i.setData(Uri.parse(url));
                startActivity(i);
            }
        });




//---------------------------Share-----Button---------------------------------
        b33.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(Intent.ACTION_SEND);
                intent.setType("text/plain");
                String Body = "Download This App";
                String Sub = "Download Free Adeeb Technology Lab chrome extension:          https://chromewebstore.google.com/detail/adeeb-technology-lab/hmgnckmfplfccnkinmmlmicpldfpnhme";
                intent.putExtra(Intent.EXTRA_TEXT, Body);
                intent.putExtra(Intent.EXTRA_TEXT, Sub);
                startActivity(Intent.createChooser(intent, "Share Using"));

            }
        });



        //---------------------------Downloader-----Button---------------------------------
        b4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String url = "https://play.google.com/store/apps/details?id=com.AdeebTechLab.AdeebTechLabSticker";
                Intent i = new Intent(Intent.ACTION_VIEW);
                i.setData(Uri.parse(url));
                startActivity(i);
            }
        });




//---------------------------Share-----Button---------------------------------
        b44.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(Intent.ACTION_SEND);
                intent.setType("text/plain");
                String Body = "Download This App";
                String Sub = "Download Free Adeeb Technology Lab Sticker App:          https://play.google.com/store/apps/details?id=com.AdeebTechLab.AdeebTechLabSticker";
                intent.putExtra(Intent.EXTRA_TEXT, Body);
                intent.putExtra(Intent.EXTRA_TEXT, Sub);
                startActivity(Intent.createChooser(intent, "Share Using"));

            }
        });



    }
}

File: app/src/main/java/com/adeebtechlab/apps/Share.kt
package com.adeebtechlab.apps

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class Share : AppCompatActivity() {

    // Navigation buttons
    private lateinit var homeButton: Button
    private lateinit var downloadButton: Button
    private lateinit var shareButton: Button
    private lateinit var profileButton: Button
    private lateinit var aboutButton: Button

    // App / Sticker / Extension / Software buttons
    private lateinit var downlord_app_bt: Button
    private lateinit var shear_app_bt: Button
    private lateinit var downlord_stiker_app_bt: Button
    private lateinit var shear_stiker_app_bt: Button
    private lateinit var downlord_extention_bt: Button
    private lateinit var shear_extention_bt: Button
    private lateinit var downlord_softwear_bt: Button
    private lateinit var shear_softwear_bt: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_share)

        // ================= Edge-to-Edge padding =================
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        // ================= Navigation buttons =================
        homeButton = findViewById(R.id.homeButton)
        downloadButton = findViewById(R.id.downloadButton)
        shareButton = findViewById(R.id.shareButton)
        profileButton = findViewById(R.id.profileButton)
        aboutButton = findViewById(R.id.aboutButton)

        homeButton.setOnClickListener { startActivity(Intent(this, Home::class.java)) }
        downloadButton.setOnClickListener { startActivity(Intent(this, Download::class.java)) }
        shareButton.setOnClickListener { startActivity(Intent(this, Share::class.java)) }
        profileButton.setOnClickListener { startActivity(Intent(this, Profile::class.java)) }
        aboutButton.setOnClickListener { startActivity(Intent(this, About::class.java)) }

        // ================= Download & Share Buttons =================
        downlord_app_bt = findViewById(R.id.downlord_app_bt)
        shear_app_bt = findViewById(R.id.shear_app_bt)
        downlord_stiker_app_bt = findViewById(R.id.downlord_stiker_app_bt)
        shear_stiker_app_bt = findViewById(R.id.shear_stiker_app_bt)
        downlord_extention_bt = findViewById(R.id.downlord_extention_bt)
        shear_extention_bt = findViewById(R.id.shear_extention_bt)
        downlord_softwear_bt = findViewById(R.id.downlord_softwear_bt)
        shear_softwear_bt = findViewById(R.id.shear_softwear_bt)

        // ================= Helper function for URL =================
        fun openUrl(url: String) {
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
            startActivity(intent)
        }

        fun shareText(message: String) {
            val intent = Intent(Intent.ACTION_SEND).apply {
                type = "text/plain"
                putExtra(Intent.EXTRA_TEXT, message)
            }
            startActivity(Intent.createChooser(intent, "Share Using"))
        }

        // ================= App Buttons =================
        downlord_app_bt.setOnClickListener {
            openUrl("https://play.google.com/store/apps/details?id=com.adeebtechlab.apps")
        }

        shear_app_bt.setOnClickListener {
            shareText("Download Free Adeeb Technology Lab App: https://play.google.com/store/apps/details?id=com.adeebtechlab.apps")
        }

        downlord_stiker_app_bt.setOnClickListener {
            openUrl("https://play.google.com/store/apps/details?id=com.AdeebTechLab.AdeebTechLabSticker")
        }

        shear_stiker_app_bt.setOnClickListener {
            shareText("Download Free Adeeb Technology Lab WhatsApp Sticker App: https://play.google.com/store/apps/details?id=com.AdeebTechLab.AdeebTechLabSticker")
        }

        downlord_extention_bt.setOnClickListener {
            openUrl("https://chrome.google.com/webstore/detail/adeeb-technology-lab/hmgnckmfplfccnkinmmlmicpldfpnhme")
        }

        shear_extention_bt.setOnClickListener {
            shareText("Download Free Adeeb Technology Lab Chrome Extension: https://chrome.google.com/webstore/detail/adeeb-technology-lab/hmgnckmfplfccnkinmmlmicpldfpnhme")
        }

        downlord_softwear_bt.setOnClickListener {
            openUrl("https://apps.microsoft.com/detail/9P3369QX7KFX?hl=en-us&gl=PK&ocid=pdpshare")
        }

        shear_softwear_bt.setOnClickListener {
            shareText("Download Free Adeeb Technology Lab Windows Software: https://apps.microsoft.com/detail/9P3369QX7KFX?hl=en-us&gl=PK&ocid=pdpshare")
        }
    }
}
